src/SFML/Graphics/Color.cpp

```diff
@@ -68,6 +68,24 @@ a(alpha)
 
 
 ////////////////////////////////////////////////////////////
+Color::Color(Uint32 color) :
+r((color & 0xff000000) >> 24),
+g((color & 0x00ff0000) >> 16),
+b((color & 0x0000ff00) >> 8 ),
+a((color & 0x000000ff) >> 0 )
+{
+
+}
+
+
+////////////////////////////////////////////////////////////
+Uint32 Color::toInteger()
+{
+    return r << 24 | g << 16 | b << 8 | a;    
```

Comment:
The rules are rather confusing with << operators: the return type is the (promoted) left operand's type which is promoted to int here because of the right operand being int. So, from my understanding of the standard it's safe as is.

But it's quite tricky and maybe I'm wrong and the compiler should emit a warning.. :grin: